src/vanflow.c

```diff
@@ -484,6 +488,15 @@ static void _vflow_start_record_TH(vflow_work_t *work, bool discard)
 
     vflow_record_t *record = work->record;
 
+    // If this is one of the record types that supports observability,
```

Comment:
This count is being done in the wrong place because this is not where the records are created.  It should be in the non-in-thread version of `vflow_start_record`.

This location is on the back-end of the work queue and thus will not protect against large bursts of records being placed in the work queue.  (Also, in its current location, there is no need for atomics).